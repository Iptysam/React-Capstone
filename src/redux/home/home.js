import { configureStore } from '@reduxjs/toolkit';

const ADDDATA = 'ADDDATA';
const URL = 'https://restcountries.com/v3.1/all';

export function Data(payload) {
  return {
    type: ADDDATA,
    payload,
  };
}

export default function Countries(state = [], action = {}) {
  switch (action.type) {
    case ADDDATA:
      return [
        ...action.payload,
      ];
    default:
      return state;
  }
}

export const fetchData = () => async (dispatch) => {
  let countryList = [];
  await fetch(URL)
    .then((response) => response.json())
    .then((data) => { countryList = data; });
  dispatch(Data(countryList));
};

export const store = configureStore({ reducer: { countries: Countries } });
