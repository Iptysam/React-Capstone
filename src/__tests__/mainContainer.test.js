import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../redux/home/home';
import CountriesContainer from '../components/mainContainer';

describe('Jest Snapshot testing suite', () => {
  it('The main countries container snapshot', () => {
    const value = renderer.create(
      <Provider store={store}>
        <Router>
          <CountriesContainer />
        </Router>
      </Provider>,
    ).toJSON();
    expect(value).toMatchSnapshot();
  });
});
