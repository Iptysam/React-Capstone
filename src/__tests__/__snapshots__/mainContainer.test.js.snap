// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite The main countries container snapshot 1`] = `
<div
  className="countriesContainer"
>
  <select
    className="continent-change"
    onChange={[Function]}
  >
    <option>
      Africa
    </option>
    <option>
      Europe
    </option>
    <option>
      Asia
    </option>
    <option>
      Antarctica
    </option>
    <option>
      North America
    </option>
    <option>
      South America
    </option>
  </select>
  <section
    className="flex"
  />
</div>
`;
